### PR TITLE
Remove disable activity and workflow worker options

### PR DESCRIPTION
**Pull request by meiliang86**
- Workers will always start, and poller will start only if types are registered.
- Simplified worker start and shutdown logic
- Fixed a bug in worker.isSuspended()

Ref: https://github.com/temporalio/sdk-java/pull/33/commits/4c1d9b02f6e73eb44d22a2e17b226611663e3373

### CODE REVIEW

File: src/main/java/com/uber/cadence/internal/worker/LocalActivityWorker.java [hunk 1]
```diff
@@ -85,46 +85,73 @@ public void start() {
 
   @Override
   public boolean isStarted() {
+    if (poller == null) {
```
Comment (vytautas-karpavicius) [inline src/main/java/com/uber/cadence/internal/worker/LocalActivityWorker.java:88]: It seems like poller is `NoopSuspendableWorker` initially (that is when it is not started). We should not need those null checks here.

Comment (meiliang86) [reply src/main/java/com/uber/cadence/internal/worker/LocalActivityWorker.java]: Good catch. Remove all of these and moved common methods implementations into an abstract base class.

File: src/main/java/com/uber/cadence/worker/WorkerOptions.java [hunk 2]
```diff
@@ -51,28 +49,6 @@
     private Scope metricsScope;
     private boolean enableLoggingInReplay;
 
-    /**
```
Comment (vancexu) [inline src/main/java/com/uber/cadence/worker/WorkerOptions.java:54]: Just a reminder we may need to mention these breaking changes in next release note.

**Review by vancexu — APPROVED** [review]

**Review by mkolodezny — APPROVED** [review]

File: src/main/java/com/uber/cadence/worker/WorkerOptions.java (hunk 2 above)
Comment (meiliang86) [reply src/main/java/com/uber/cadence/worker/WorkerOptions.java:54]: Yes. The next release will be 3.0

**Pull request merged**